### DISCUSSION TITLE
Text/Paragraph component

### DISCUSSION THREAD

**Discussion by LasseRafn** [livewire/flux#698] [Feature Requests] [answered]
Flux provides headings, subheadings etc. with set colors for both light and dark mode, but whenever I need to add a text / paragraph below, I am out of options, so I do something like this:

```html
<p class="text-black dark:text-white">my text</p>
```

Its a silly request, but I wonder if something like:

```html
<flux:paragraph>my text</flux:paragraph>
```

or

```html
<flux:text>my text</flux:text>
```

would make sense? with some predefined colors etc.

I don't mind doing it myself, but feel like it would be a 'cheap-to-maintain' component to add some extra value to the framework.

Just an idea, feel free to yeet it away

**Comment by jeffchown** [comment]
@LasseRafn There is a `flux:text` component already (not documented yet) that I use for exactly the reason you mention.

**Reply by jeffchown** [reply, marked as answer]
For reference, this is its blade code so you can see the `size` options and the default styling:

```
@props([
    'size' => null,
])

@php
$classes = Flux::classes()
    ->add(match ($size) {
        'xl' => 'text-lg',
        'lg' => 'text-base',
        default => 'text-sm',
        'sm' => 'text-xs',
    })
    ->add('text-zinc-500 dark:text-white/70')
    ;
@endphp

<div {{ $attributes->class($classes) }} data-flux-text>
    {{ $slot }}
</div>
```

**Reply by LasseRafn** [reply]
Oh that's perfect @jeffchown !! Thanks 🙏🏻

**Reply by jeffchown** [reply]
You're welcome @LasseRafn !
